scripts/zpool

```diff
@@ -47,28 +47,40 @@ This is a resource agent that manages ZFS zpools.
 EOF
 }
 
+zpool_status_log() {
+    # Log zpool status to splunk for post-mortem.  Run in it the background
+    # in case zpool status hangs.  Wait two seconds for it to log.
+    nohup sh -c 'zpool status | logger -t resource_agent_fail -p local7.debug' &
+    sleep 2
+}
+
 zpool_monitor() {
     local rc
+    rc=$OCF_NOT_RUNNING
 
     # exit immediately if configuration is not valid
     zpool_validate_all || exit $?
 
-    ocf_run -q -info zpool status $OCF_RESKEY_pool
+    if [ -e /proc/spl/kstat/zfs/$OCF_RESKEY_pool/state ] ; then
+        # Best case, we can read the pool health locklessly from the proc file.
+        state=$(cat /proc/spl/kstat/zfs/$OCF_RESKEY_pool/state)
 
-    # This example assumes the following exit code convention
-    # for zpool status:
-    # 0: imported
-    # any other: not imported
-    case "$?" in
-        0)
+        if [ "$state" == "SUSPENDED" ] || [ "$state" == "FAULTED" ] || [ "$state" == "UNAVAIL" ] ; then
```

Comment:
"==" is a bug, they should be "=".  posix sh [] and test only support a single equal sign.

Also, this could be one conditional:

if [ "$state" = "SUSPENDED" -o "$state" = "FAULTED" -o "$state" = "UNAVAIL" ] ; then

Reply:
Good point, I'll fix